garbage_classifier/train.py

```diff
@@ -100,21 +27,30 @@ def train_model(config=None):
         val_dataloader = garbage_data.get_val_loader()
 
         # model
-        model = get_model(config.dropout, config.fc_layer_size)
+        model = get_model(config.dropout, config.fc_layer_size, opt.classes)
+        model.to(opt.device)
 
         # optimizer
         optimizer = get_optimizer(model, config.optimizer, config.lr)
         f1_score = 0
-
+        avg_loss = 0
         for epoch in range(config.epochs):
-            avg_loss = train_epoch(epoch, model, train_dataloader, optimizer)
+            print(f'Epoch {epoch} started...')
+            avg_loss = train_epoch(model, train_dataloader, optimizer, opt.device)
             wandb.log({"loss": avg_loss})
-            f1_score = score_model(model, val_dataloader)
+            f1_score = score_model(model, val_dataloader, opt.device)
             wandb.log({'val_f1': f1_score})
             torch.save(model, join(opt.checkpoint_dir,
                        f'epoch{epoch}_f1={round(f1_score, 5)}.pth'))
         # create card
         create_card(config, f1_score)
+        # save model
+        torch.save(model, join(opt.checkpoint_dir, 'weights', 'latest', 'model.pth'))
+        art = wandb.Artifact(f'uwg-garbage-classifier-{wandb.run.id}', type="model")
+        art.add_file(join(opt.checkpoint_dir, 'weights', 'latest', 'model.pth'), "model.pth")
+        art.add_file("garbage_classifier/README.md", "README.md")
```

Comment:
could we make this a function?

Reply:
done in further prs :)

https://github.com/rosklyar/projector-ml-ops/pull/10/files#diff-d63aebfeb03543fee5807276a22ea2ed95c38661d0613ce71858b1fa4337e86c